src/App.tsx

```diff
@@ -1,26 +1,187 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useRef, useState } from 'react';
 
-const USER_ID = 0;
+import { UserWarning } from './UserWarning';
+import { USER_ID, getTodos, postTodos, deleteTodos } from './api/todos';
+import { TodoList } from './Components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { Footer } from './Components/Footer/Footer';
+import { FilterStatus } from './types/FilterStatus';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+import { ErrorNotification } from './Components/ErrorNotification';
+import { getCountOfCompletedTodos } from './utils/getCountOfCompletedTodos';
+import { ErrorMessage } from './types/ErrorMessages';
 
 export const App: React.FC = () => {
+  const [newTitle, setNewTitle] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterStatus>(FilterStatus.All);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isAllDeleting, setIsAllDeleting] = useState<boolean>(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handlerErrorShow = (error: ErrorMessage): void => {
+    setErrorMessage(error);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        handlerErrorShow(ErrorMessage.LoadError);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setNewTitle('');
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isLoading]);
+
+  const handlerAddTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    const normalizedTitle = newTitle.trim();
+
+    if (!normalizedTitle) {
+      handlerErrorShow(ErrorMessage.TitleError);
+      setIsLoading(false);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: todos.length + 1,
+      completed: false,
+      userId: USER_ID,
+      title: normalizedTitle,
+    };
+
+    const temporaryTodo = { ...newTodo, id: 0 };
+
+    setTempTodo(temporaryTodo);
+
+    postTodos(newTodo)
+      .then(() => {
+        setTodos([...todos, newTodo]);
```

Comment:
Use functional `setState` when you need to update it depending on the previous one